.github/workflows/release.yaml

```diff
@@ -0,0 +1,578 @@
+name: Release process
+on:
+  # Releases can be manually triggered if the version has already been updated and the apidocs were
+  # properly built
+  workflow_dispatch:
+
+  # Releases are also allowed to be automatically triggered once the release preparation 
+  # pull-request (targeting main) is merged. Here, we also make sure that the PR has changes 
+  # in at least one file related to `make set_version` or `make apidocs`
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+    paths:
+      - src/concrete/ml/version.py
+      - 'docs/developer-guide/api/**'
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: false
+
+env:
+  ACTION_RUN_URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}
+
+jobs:
+  # Open the AWS EC2 instance if the release process has been triggered manually or if the release 
+  # preparation pull-request has been merged
+  # That latter PR is found by checking for closed PR merged (into main) that contains "Prepare 
+  # release" in their title coming from a branch that starts with "chore/prepare_release_"
+  start-runner-linux:
+    name: Start EC2 runner
+    if: |
+      github.event_name == 'workflow_dispatch'
+      || (
+        github.event_name == 'pull_request'
```

Comment:
if the workflow has been triggered automatically by a closed PR, then we will check that the PR has been merged (and not just closed), came from a branch starting by "chore/prepare_release_" and contains "Prepare release" in the title. If it does not, all following jobs are skipped. 

I think these restrictions are enough for the automatic trigger (and does not open many runners as the first filter is very restrictive). However, I understand that we might want to remove it (and then assume that we'll manually trigger the release process every time after the prepare release one)